src/Layout/redist/targets/GenerateLayout.targets

```diff
@@ -4,55 +4,59 @@
     <PackageDownload Include="Microsoft.NETCore.Platforms" Version="[$(MicrosoftNETCorePlatformsPackageVersion)]" />
   </ItemGroup>
 
-  <Target Name="PublishPortableRuntimeIdentifierGraph">
+  <Target Name="PublishRuntimeIdentifierGraphFiles">
+    <PropertyGroup>
+      <NETCoreSdkRuntimeIdentifierChainTargetFile>$(OutputPath)NETCoreSdkRuntimeIdentifierChain.txt</NETCoreSdkRuntimeIdentifierChainTargetFile>
+      <PortableRuntimeIdentifierGraphTargetFile>$(OutputPath)PortableRuntimeIdentifierGraph.json</PortableRuntimeIdentifierGraphTargetFile>
+    </PropertyGroup>
+
     <Copy SourceFiles="$(NuGetPackageRoot)/microsoft.netcore.platforms/$(MicrosoftNETCorePlatformsPackageVersion)/runtime.json"
           DestinationFiles="$(OutputPath)RuntimeIdentifierGraph.json"
           SkipUnchangedFiles="true" />
 
     <Copy SourceFiles="$(NuGetPackageRoot)/microsoft.netcore.platforms/$(MicrosoftNETCorePlatformsPackageVersion)/PortableRuntimeIdentifierGraph.json"
-          DestinationFolder="$(OutputPath)"
+          DestinationFiles="$(PortableRuntimeIdentifierGraphTargetFile)"
           SkipUnchangedFiles="true" />
+  </Target>
 
+  <Target Name="GenerateSdkRuntimeIdentifierChain"
+          DependsOnTargets="PublishRuntimeIdentifierGraphFiles"
+          Inputs="$(PortableRuntimeIdentifierGraphTargetFile)"
+          Outputs="$(NETCoreSdkRuntimeIdentifierChainTargetFile)">
     <GenerateSdkRuntimeIdentifierChain
-          RuntimeIdentifier="$(PortableProductMonikerRid)"
-          RuntimeIdentifierGraphPath="$(OutputPath)PortableRuntimeIdentifierGraph.json"
-          RuntimeIdentifierChainOutputPath="$(OutputPath)NETCoreSdkRuntimeIdentifierChain.txt" />
+      RuntimeIdentifier="$(PortableProductMonikerRid)"
+      RuntimeIdentifierGraphPath="$(PortableRuntimeIdentifierGraphTargetFile)"
+      RuntimeIdentifierChainOutputPath="$(NETCoreSdkRuntimeIdentifierChainTargetFile)" />
   </Target>
 
   <Target Name="PublishVersionFile">
     <WriteLinesToFile File="$(OutputPath).toolsetversion"
                       Lines="$(SourceRevisionId);$(Version);$(TargetRid)"
-                      Overwrite="true" />
+                      Overwrite="true"
+                      WriteOnlyWhenDifferent="true" />
 
     <WriteLinesToFile File="$(OutputPath).version"
                       Lines="$(SourceRevisionId);$(Version);$(TargetRid);$(FullNugetVersion);$(SdkFeatureBand)"
-                      Overwrite="true" />
-
-    <!-- Development env and CI loads DotnetFiles type from the following location
-         The resolution of the product version (https://github.com/dotnet/sdk/blob/main/src/Cli/Microsoft.DotNet.Cli.Utils/DotnetFiles.cs#L21)
```

Comment:
Why is this not needed anymore?

Reply:
This was necessary before sdk and installer got merged as that file wasn't available in sdk. I will double check this locally tomorrow.

Reply:
Double-checked? If so, I think this is good to go

Reply:
Just verified, yes this change is fine. This is already handled by the above WriteLinesToFile task with the OverlaySdkOnLkg.targets.